app/services/time_slot_computer.rb

```diff
@@ -0,0 +1,50 @@
+class TimeSlotComputer
+  CLOCK_TIME_REGEX = /^\s?(\d{1,2})\s?:\s?(\d{1,2})\s?([ap]m)\s?/i.freeze
+
+  attr_reader :date, :duration, :start_clock_time, :end_clock_time
+
+  def initialize(date, start_clock_time, end_clock_time, duration_minutes = 60)
+    # *_clock_time are strings like "11:00 PM" or "5:00 am"
+    @duration = duration_minutes
+    @date = date
+    @start_clock_time = start_clock_time
+    @end_clock_time = end_clock_time
+  end
+
+  def run
+    return [] unless date && start_clock_time && end_clock_time
+
+    start_time = build_date_time(date, start_clock_time)
+    end_time = build_date_time(date, end_clock_time)
+    current = start_time
+    slots = []
+    while current < end_time
+      current_end = current + duration.minutes
+      slots << format_slot(current, current_end)
+      current = current_end
+    end
+    slots
```

Comment:
I feel like this could/ should be rewritten as a `reduce`, but I get it and this accomplishes what it needs to. 😅

Reply:
I didn't go straight to reduce because i didn't have an array of start elements (easily) and to construct that i wanted to do the `+=` in case the start/end time are not divisible by `duration.minutes` but i get it and will give it a try with reduce.    or i might find you and spitball around it.